### PR TITLE
Suspenders! (And subsequent recipes.)

### PR DESCRIPTION
**Pull request by slimeboy460**
#### Summary
None

#### Purpose of change

This is a content update that adds three new times of suspenders. Cotton, Leather, and Holster.

#### Describe the solution

This was simply developed to add alternate forms of belts. For players who want their character to have more of a western or victorian era vibe to them.

#### Describe alternatives you've considered

N/A

#### Testing

I ran it through the baseline tests, using JSONlint and the baseline tests. I think it should all be clear but if there are any more issues I'll fix them if I notice them!

#### Additional context

The recipes might be due to some tuning, more for the cotton suspenders than anything. The leather Suspenders and Holster Suspenders should be good, however!

### CODE REVIEW

File: data/json/recipes/armor/torso.json [hunk 1]
```diff
@@ -2456,5 +2456,33 @@
...
-  }
+  },
+  {
+    "result": "suspenders_cotton",
+    "type": "recipe",
+    "activity_level": "LIGHT_EXERCISE",
+    "category": "CC_ARMOR",
+    "subcategory": "CSC_ARMOR_TORSO",
+    "skill_used": "tailor",
+    "difficulty": 1
```
Comment (BrettDong) [inline data/json/recipes/armor/torso.json:2467]:
```suggestion
    "difficulty": 1,
```

File: data/json/recipes/armor/torso.json [hunk 2]
```diff
@@ -2456,5 +2456,33 @@
...
+    "proficiencies": [ { "proficiency": "prof_closures" } ]
+  },
+  {
+    "result": "suspenders_leather",
+    "type": "recipe",
+    "activity_level": "LIGHT_EXERCISE",
+    "category": "CC_ARMOR",
+    "subcategory": "CSC_ARMOR_TORSO",
+    "skill_used": "tailor",
+    "difficulty": 1
```
Comment (BrettDong) [inline data/json/recipes/armor/torso.json:2481]:
```suggestion
    "difficulty": 1,
```

File: data/json/recipes/armor/storage.json [hunk 3]
```diff
@@ -1476,5 +1476,19 @@
...
     "components": [ [ [ "cordage_short", 1, "LIST" ] ] ]
+  },
+  {
+    "result": "suspenders_holster",
+    "type": "recipe",
+    "activity_level": "LIGHT_EXERCISE",
+    "category": "CC_ARMOR",
+    "subcategory": "CSC_ARMOR_STORAGE",
+    "skill_used": "tailor",
+    "difficulty": 3
```
Comment (BrettDong) [inline data/json/recipes/armor/storage.json:1487]:
```suggestion
    "difficulty": 3,
```

File: data/json/recipes/armor/torso.json [hunk 4]
```diff
@@ -2456,5 +2456,33 @@
...
+    "category": "CC_ARMOR",
+    "subcategory": "CSC_ARMOR_TORSO",
+    "skill_used": "tailor",
+    "difficulty": 1,
+    "time": "2 h",
+    "autolearn": true,
+    "reversible": true,
+    "using": [ [ "tailoring_leather_small", 4 ], [ "clasps", 3 ] ],
+    "proficiencies": [ { "proficiency": "prof_closures" }, { "proficiency": "prof_leatherworking_basic" } ]
+  },
```
Comment (BrettDong) [inline data/json/recipes/armor/torso.json:2487]:
```suggestion
  }
```

**Comment by Maleclypse** [comment]
This needs a rebase and linting. Do you need any help with that?

**Comment by slimeboy460** [comment]
> This needs a rebase and linting. Do you need any help with that?

I can do the linting but not the rebase, I kinda got caught up with some things irl but I'll try and do the linting by tomorrow

**Comment by slimeboy460** [comment]
Noticed the problem, i accidentally stuck commas where the EOF should've been, which I think was causing the conflicts

File: data/json/items/armor/belts.json [hunk 5]
```diff
@@ -297,5 +297,41 @@
     "use_action": { "type": "holster", "holster_prompt": "Store tool or blade", "holster_msg": "You put your %1$s in your %2$s" },
     "flags": [ "WAIST", "WATER_FRIENDLY" ],
     "armor": [ { "coverage": 5, "covers": [ "torso" ] } ]
+  },
+  {
+    "id": "suspenders_cloth",
+    "type": "ARMOR",
+    "name": { "str": "Suspenders" },
```
Comment (TheShadowFerret) [inline data/json/items/armor/belts.json:304]:
```suggestion
    "name": { "str": "suspenders", "str_pl": "pairs of suspenders" },
```

File: data/json/items/armor/belts.json [hunk 6]
```diff
@@ -297,5 +297,41 @@
...
+    "looks_like": "leather_belt",
+    "color": "black",
+    "material_thickness": 1.5,
+    "flags": [ "WAIST", "WATER_FRIENDLY" ], 
+    "armor": [ { "coverage": 5, "covers": [ "torso" ] } ]
+  },
+  {
+    "id": "suspenders_leather",
+    "type": "ARMOR",
+    "name": { "str": "Leather Suspenders" },
```
Comment (TheShadowFerret) [inline data/json/items/armor/belts.json:321]:
```suggestion
    "name": { "str": "leather suspenders", "str_pl": "pairs of leather suspenders" },
```

File: data/json/items/armor/belts.json [hunk 7]
```diff
@@ -297,5 +297,41 @@
...
+    "description": "A pair of leather suspenders, which reach over the wearer's shoulders and hook to their waist. Gives a wonderful western feel!",
+    "material": [ "leather", "steel" ],
+    "color": "brown",
+    "copy_from": "suspenders_cloth"
+  },
+  {
+    "id": "suspenders_holster",
+    "type": "ARMOR",
+    "name": { "str": "Holster Suspenders" },
```
Comment (TheShadowFerret) [inline data/json/items/armor/belts.json:330]:
```suggestion
    "name": { "str": "holster suspenders", "str_pl": "pairs of holster suspenders" },
```

**Review by TheShadowFerret — CHANGES_REQUESTED** [review]
Plurals + no caps

**Comment by slimeboy460** [comment]
Thank you for that, sorry about not replying sooner, College kicked my ass with a ten page paper

File: data/json/items/armor/belts.json [hunk 8]
```diff
@@ -337,5 +337,50 @@
...
+    "weight": "250g",
+    "volume": "500 ml",
+    "price": 1000,
+    "price_postapoc": 25,
+    "material": [ "cotton", "steel" ],
+    "symbol": "#",
+    "looks_like": "leather_belt",
+    "color": "black",
+    "material_thickness": 1.5,
+    "flags": [ "WAIST", "WATER_FRIENDLY" ],
```
Comment (dseguin) [inline data/json/items/armor/belts.json:355]:
```suggestion
    "flags": [ "WAIST", "WATER_FRIENDLY" ],
```

File: data/json/items/armor/belts.json [hunk 9]
```diff
@@ -337,5 +337,50 @@
...
+    "armor": [ { "coverage": 5, "covers": [ "torso" ] } ]
```
Comment (Termineitor244) [inline data/json/items/armor/belts.json:356]:
```suggestion    
    "flags": [ "BELTED", "WATER_FRIENDLY" ],
    "armor": [ { "coverage": 50, "covers": [ "torso" ], "specifically_covers": [ "torso_waist" ] } ]
```

File: data/json/items/armor/belts.json [hunk 10]
```diff
@@ -337,5 +337,50 @@
...
+    "copy_from": "suspenders_cloth"
+  },
+  {
+    "id": "suspenders_holster",
+    "type": "ARMOR",
+    "name": { "str": "holster suspenders", "str_pl": "pairs of holster suspenders" },
+    "description": "A pair of leather suspenders with a holster placed near the stomach. It was designed with subtlety in mind so it can hold no more than a small revolver.",
+    "copy_from": "suspenders_leather",
+    "use_action": { "type": "holster", "holster_prompt": "Store tool or blade", "holster_msg": "You put your %1$s in your %2$s" },
+    "flags": [ "WAIST", "WATER_FRIENDLY", "NO_QUICKDRAW" ],
```
Comment (Termineitor244) [inline data/json/items/armor/belts.json:374]:
```suggestion
    "flags": [ "BELTED", "WATER_FRIENDLY", "NO_QUICKDRAW" ],
```

File: data/json/items/armor/belts.json [hunk 11]
```diff
@@ -337,5 +337,50 @@
...
+    "pocket_data": [
+      {
+        "holster": true,
+        "max_contains_volume": "500 ml",
+        "max_contains_weight": "3000 g",
+        "max_item_length": "20 cm",
+        "moves": 60,
+        "flag_restriction": [ "BELT_CLIP" ]
+      }
+    ]
```
Comment (Termineitor244) [inline data/json/items/armor/belts.json:384]:
```suggestion
    "pocket_data": [
      {
        "magazine_well": "350 ml",
        "pocket_type": "CONTAINER",
        "holster": true,
        "min_item_volume": "250 ml",
        "max_contains_volume": "1000 ml",
        "max_contains_weight": "2 kg",
        "max_item_length": "30 cm",
        "moves": 50
      }
    ],
    "use_action": { "type": "holster" },
    "armor": [ { "encumbrance": [ 3, 4 ], "coverage": 75, "covers": [ "torso" ], "specifically_covers": [ "torso_waist" ] } ]
  }
```

File: data/json/items/armor/belts.json [hunk 12]
```diff
@@ -337,5 +337,50 @@
...
+    "description": "A pair of leather suspenders, which reach over the wearer's shoulders and hook to their waist. Gives a wonderful western feel!",
+    "material": [ "leather", "steel" ],
+    "color": "brown",
+    "copy_from": "suspenders_cloth"
+  },
+  {
+    "id": "suspenders_holster",
+    "type": "ARMOR",
+    "name": { "str": "holster suspenders", "str_pl": "pairs of holster suspenders" },
+    "description": "A pair of leather suspenders with a holster placed near the stomach. It was designed with subtlety in mind so it can hold no more than a small revolver.",
```
Comment (Termineitor244) [inline data/json/items/armor/belts.json:371]:
```suggestion
    "description": "A pair of leather suspenders with a holster placed near the stomach.",
```

**Review by Termineitor244 — CHANGES_REQUESTED** [review]
I adjusted your items to use the new "torso_waist" sublocation and your holster suspender to use corrected values for its pocket (since you are building it with a normal holster) and added some encumbrance.

You need to add these items to some spawn lists if you want them to be found in game and not only crafted.

File: data/json/items/armor/belts.json (hunk 10 above)
Comment (Termineitor244) [reply data/json/items/armor/belts.json:374]: I'm not sure if you really want "NO_QUICKDRAW" for this item too.

File: data/json/items/armor/belts.json (hunk 11 above)
Comment (Termineitor244) [reply data/json/items/armor/belts.json:384]: If you want another value for its pocket and encumbrance (And to retain its original description) then you need to build the item with another, smaller holster.

File: data/json/items/armor/belts.json [hunk 13]
```diff
@@ -337,5 +337,55 @@
...
+        "min_item_volume": "250 ml",
+        "max_contains_volume": "1000 ml",
+        "max_contains_weight": "2 kg",
+        "max_item_length": "30 cm",
+        "moves": 50
+      }
+    ],
+    "use_action": { "type": "holster" },
+    "armor": [ { "encumbrance": [ 3, 4 ], "coverage": 75, "covers": [ "torso" ], "specifically_covers": [ "torso_waist" ] } ]
+  }
```
Comment (dseguin) [inline data/json/items/armor/belts.json:389]:
```suggestion
```

File: data/json/items/armor/belts.json [hunk 14]
```diff
@@ -337,5 +337,54 @@
...
+        "pocket_type": "CONTAINER",
+        "holster": true,
+        "min_item_volume": "250 ml",
+        "max_contains_volume": "1000 ml",
+        "max_contains_weight": "2 kg",
+        "max_item_length": "30 cm",
+        "moves": 50
+      }
+    ],
+    "use_action": { "type": "holster" },
```
Comment (dseguin) [inline data/json/items/armor/belts.json:387]:
```suggestion
```

File: data/json/items/armor/belts.json [hunk 15]
```diff
@@ -337,5 +337,54 @@
...
+    "material": [ "leather", "steel" ],
+    "color": "brown",
+    "copy_from": "suspenders_cloth"
+  },
+  {
+    "id": "suspenders_holster",
+    "type": "ARMOR",
+    "name": { "str": "holster suspenders", "str_pl": "pairs of holster suspenders" },
+    "description": "A pair of leather suspenders with a holster placed near the stomach.",
+    "copy_from": "suspenders_leather",
```
Comment (dseguin) [inline data/json/items/armor/belts.json:372]:
```suggestion
    "copy-from": "suspenders_leather",
```

File: data/json/items/armor/belts.json [hunk 16]
```diff
@@ -337,5 +337,54 @@
...
+    "armor": [ { "coverage": 50, "covers": [ "torso" ], "specifically_covers": [ "torso_waist" ] } ]
+  },
+  {
+    "id": "suspenders_leather",
+    "type": "ARMOR",
+    "name": { "str": "leather suspenders", "str_pl": "pairs of leather suspenders" },
+    "description": "A pair of leather suspenders, which reach over the wearer's shoulders and hook to their waist. Gives a wonderful western feel!",
+    "material": [ "leather", "steel" ],
+    "color": "brown",
+    "copy_from": "suspenders_cloth"
```
Comment (dseguin) [inline data/json/items/armor/belts.json:365]:
```suggestion
    "copy-from": "suspenders_cloth"
```

File: data/json/recipes/armor/torso.json [hunk 17]
```diff
@@ -2631,5 +2631,27 @@
...
+    "skill_used": "tailor",
+    "difficulty": 1,
+    "time": "1 h",
+    "autolearn": true,
+    "reversible": true,
+    "using": [ [ "rag", 5 ], [ "clasps", 3 ] ],
+    "proficiencies": [ { "proficiency": "prof_closures" } ]
+  },
+  {
+    "type": "recipe",
```
Comment (dseguin) [inline data/json/recipes/armor/torso.json:2647]:
```suggestion
    "type": "recipe",
    "activity_level": "LIGHT_EXERCISE",
    "category": "CC_ARMOR",
    "subcategory": "CSC_ARMOR_TORSO",
```

File: data/json/recipes/armor/torso.json [hunk 18]
```diff
@@ -2631,5 +2631,27 @@
     "qualities": [ { "id": "CUT", "level": 1 } ],
     "components": [ [ [ "bag_garbage", 1 ] ] ],
     "flags": [ "BLIND_HARD" ]
+  },
+  {
+    "type": "recipe",
+    "result": "suspenders_cotton",
```
Comment (dseguin) [inline data/json/recipes/armor/torso.json:2637]:
```suggestion
    "type": "recipe",
    "activity_level": "LIGHT_EXERCISE",
    "category": "CC_ARMOR",
    "subcategory": "CSC_ARMOR_TORSO",
    "result": "suspenders_cloth",
```

File: data/json/recipes/armor/torso.json [hunk 19]
```diff
@@ -2631,5 +2631,27 @@
...
+    "skill_used": "tailor",
+    "difficulty": 1,
+    "time": "1 h",
+    "autolearn": true,
+    "reversible": true,
+    "using": [ [ "rag", 5 ], [ "clasps", 3 ] ],
```
Comment (dseguin) [inline data/json/recipes/armor/torso.json:2643]:
```suggestion
    "using": [ [ "tailoring_cotton", 1 ], [ "clasps", 3 ], [ "fabric_standard_nostretch", 5 ] ],
```

Not super sure about this one, but "rag" is not a valid requirement. It also needs some kind of tool requirement ("tailoring_cotton").

File: data/json/recipes/armor/storage.json [hunk 20]
```diff
@@ -1750,5 +1750,19 @@
...
+    "type": "recipe",
+    "activity_level": "LIGHT_EXERCISE",
+    "category": "CC_ARMOR",
+    "subcategory": "CSC_ARMOR_STORAGE",
+    "skill_used": "tailor",
+    "difficulty": 3,
+    "time": "2 h 30 m",
+    "autolearn": true,
+    "reversible": true,
+    "using": [ [ "tailoring_leather_small", 4 ], [ "clasps", 3 ], [ "holster", 1 ] ],
```
Comment (dseguin) [inline data/json/recipes/armor/storage.json:1765]:
```suggestion
    "using": [ [ "tailoring_leather_small", 4 ], [ "clasps", 3 ] ],
    "components": [ [ [ "holster", 1 ] ] ],
```

**Pull request merged**